grails-app/views/order/_filters.gsp

```diff
@@ -41,26 +41,40 @@
-            </div>
-            <div class="filter-list-item">
-                <label><warehouse:message code="order.destinationParty.label"/></label>
-                <g:selectOrganization name="destinationParty"
-                                      id="destinationParty"
-                                      value="${params.destinationParty}"
-                                      roleTypes="[org.pih.warehouse.core.RoleType.ROLE_BUYER]"
-                                      noSelection="['':'']"
-                                      class="chzn-select-deselect"
-                                      disabled="${isCentralPurchasingEnabled}" />
-            </div>
+            <g:if test="${params.orderTypeCode == OrderTypeCode.PURCHASE_ORDER}">
+                <div class="filter-list-item">
+                    <label><warehouse:message code="order.destination.label"/></label>
+                    <g:selectLocationViaAjax id="destination"
+                                             name="destination"
+                                             class="ajaxSelect2"
+                                             noSelection="['':'']"
+                                             value="${params.destination}"
+                                             data-ajax--url="${request.contextPath }/json/findLocations?activityCode=${org.pih.warehouse.core.ActivityCode.RECEIVE_STOCK}"
+                                             data-allow-clear="true"/>
+                </div>
+                <div class="filter-list-item">
+                    <label><warehouse:message code="order.destinationParty.label"/></label>
+                    <g:selectOrganization name="destinationParty"
+                                          id="destinationParty"
+                                          value="${params.destinationParty}"
+                                          roleTypes="[org.pih.warehouse.core.RoleType.ROLE_BUYER]"
+                                          noSelection="['':'']"
+                                          class="chzn-select-deselect"
+                                          disabled="${isCentralPurchasingEnabled}" />
+                </div>
+            </g:if>
+            <g:elseif test="${params.orderTypeCode == OrderTypeCode.TRANSFER_ORDER}">
+                <div class="filter-list-item">
+                    <label><warehouse:message code="order.destination.label"/></label>
+                    <g:selectLocationViaAjax id="destination"
+                                             name="destination"
+                                             class="ajaxSelect2"
+                                             noSelection="['':'']"
+                                             value="${params.destination}"
+                                             data-ajax--url="${request.contextPath }/json/findLocations?activityCode=${org.pih.warehouse.core.ActivityCode.PLACE_ORDER}"
```

Comment:
Why PLACE_ORDER for transfer order?

Reply:
It used to work like that before for both purchase order and transfer order(putaway)

Reply:
@jmiranda Do you think that it should be changed and not filtered by any activity code?

Reply:
> It used to work like that before for both purchase order and transfer order(putaway)

Oh right. Sorry about that. 

> Do you think that it should be changed and not filtered by any activity code?

Good question. I'm not sure. We can leave it as-is for now since no one has complained.